File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] ans = new int[nums.length];
        int ind = ans.length - 1;
        
        int left = 0;
        int right = nums.length - 1;
        while(left <= right) {
            if(Math.pow(nums[left], 2) > Math.pow(nums[right], 2))
                ans[ind--] = (int)Math.pow(nums[left++], 2);
            else
                ans[ind--] = (int)Math.pow(nums[right--], 2);
        }
        return ans;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> ans(nums.size());
        int ind = nums.size() - 1;
        
        int left = 0;
        int right = nums.size() - 1;
        while(left <= right) {
            if(pow(nums[left], 2) > pow(nums[right], 2))
                ans[ind--] = pow(nums[left++], 2);
            else
                ans[ind--] = pow(nums[right--], 2);
        }
        return ans;
    }
};